### PR TITLE
New merge strategy

### PR DESCRIPTION
**Pull request by domaspoliakas**
Not for merging

Since I had to test a lot on master - the changes are scattered all over the place. So to make the review process easier - here's a PR from master to a point on master before the new-merge-strategy stuff. This should include all the relevant changes. Any comments made on this I'll fix in a separate PR as a result.

I have tested that:
- auto-merge works for. branches that start with `version-bump` and that are tagged with `version: revision`
- merging with the button works for everything else, i.e. we end up with a bumped version, and a publish having executed
- a version label is correctly expected (🛑 also works as before)

So I think this should be ready for rollout.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -116,7 +98,7 @@ jobs:
           rm targets.tar
 
       - name: Common sbt setup
-        run: $SBT ++${{ matrix.scala }} transferCommonResources transferPublishAndTagResources
+        run: $SBT transferCommonResources transferPublishAndTagResources
 
       - run: ./scripts/commonSetup
```
Comment (jsantos17) [inline .github/workflows/ci.yml:104]: From what I gather here, we will now merge the PR and then a Github action will notice the merge and push a version bump based on the PR labels. 

So now I wonder if we should restrict `publishAndTag` to commits generated by Github actions.

Comment (domaspoliakas) [reply .github/workflows/ci.yml:104]: I have a condition on the publish action to only run on version release commits

Comment (jsantos17) [reply .github/workflows/ci.yml:104]: Ah I see no longer use our own scripts to publish

**Review by jsantos17 — APPROVED** [review]

**Comment by domaspoliakas** [comment]
Closing in favour of https://github.com/precog/sbt-precog-plugin/pull/12

**Pull request closed without merging**